File: pinyougou-web/pinyougou-manager-web/src/main/java/com/pinyougou/manager/controller/GoodsController.java
package com.pinyougou.manager.controller;

import com.alibaba.dubbo.config.annotation.Reference;
import com.pinyougou.common.pojo.PageResult;
import com.pinyougou.pojo.Goods;
import com.pinyougou.service.GoodsService;
import org.apache.commons.lang3.StringUtils;
import org.springframework.security.core.context.SecurityContext;
import org.springframework.security.core.context.SecurityContextHolder;
import org.springframework.web.bind.annotation.*;

@RestController
@RequestMapping("/goods")
public class GoodsController {
    @Reference(timeout = 10000)
    private GoodsService goodsService;
    @GetMapping("/findByPage")
    public PageResult findByPage(Goods goods,Integer page,Integer rows){
         //设置状态码为0
          goods.setAuditStatus("0");
        try{
            if (goods.getGoodsName() != null && StringUtils.isNoneBlank(goods.getGoodsName())){
                goods.setGoodsName(new String(goods.getGoodsName().getBytes("ISO8859-1"),"UTF-8"));
            }
        }catch (Exception e){
            e.printStackTrace();
        }
        return goodsService.findByPage(goods,page,rows);
    }
    @GetMapping("/updateStatus")
    public boolean updateStatus(Long[] ids,String status){
        try {
            goodsService.updateStatus("audit_status",ids,status);
            return true;
        } catch (Exception e) {
            e.printStackTrace();
        }
        return false;
    }
    @GetMapping("/delete")
    public boolean delete(Long[] ids) {
        try {
            goodsService.updateStatus("is_delete", ids,"1");
            return true;
        } catch (Exception e) {
            e.printStackTrace();
        }
        return false;
    }
}

File: pinyougou-web/pinyougou-manager-web/src/main/webapp/js/controller/goodsController.js
/** 定义控制器层 */
app.controller('goodsController', function($scope, $controller, baseService){

    /** 指定继承baseController */
    $controller('baseController',{$scope:$scope});

    /** 查询条件对象 */
    $scope.searchEntity = {};
    /** 分页查询(查询条件) */
    $scope.search = function(page, rows){
        baseService.findByPage("/goods/findByPage", page,
			rows, $scope.searchEntity)
            .then(function(response){
                /** 获取分页查询结果 */
                $scope.dataList = response.data.rows;
                /** 更新分页总记录数 */
                $scope.paginationConf.totalItems = response.data.total;
            });
    };

    /** 添加或修改 */
    $scope.updateStatus = function(status){
        baseService.sendGet("/goods/updateStatus?ids=" + $scope.ids + "&status=" + status)
            .then(function(response){
                if (response.data){
                    /** 重新加载数据 */
                    $scope.reload();
                    //清空ids
                    $scope.ids = [];
                }else{
                    alert("操作失败！");
                }
            });
    };

    /** 批量删除 */
    $scope.delete = function(){
        if ($scope.ids.length > 0){
            baseService.sendGet("/goods/delete?ids="+ $scope.ids)
                .then(function(response){
                    if (response.data){
                        /** 重新加载数据 */
                        $scope.reload();
                        $scope.ids = [];
                    }else{
                        alert("删除失败！");
                    }
                });
        }else{
            alert("请选择要删除的记录！");
        }
    };
});